envoy/stats/histogram.h

```diff
@@ -69,6 +69,12 @@ class HistogramStatistics {
    */
   virtual const std::vector<uint64_t>& computedBuckets() const PURE;
 
+  /**
+   * Returns nonoverlapping version of computedBuckets(). This vector is
+   * guaranteed to be the same length as supportedBuckets().
+   */
+  virtual const std::vector<uint64_t> nonoverlappingComputedBuckets() const PURE;
```

Comment:
2 nits :)

1. https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html -- you can just return the vector without making it const in the declaration -- that's a no-op
2. rename new function to computeNonOverlappingBuckets(). The function is doing the computation and returning the result, not returning an already-existing result, so I think it's better to start the function name with the verb.

Reply:
sorry that was the wrong clang link; it's this one that's applicable: https://clang.llvm.org/extra/clang-tidy/checks/readability-const-return-type.html

Reply:
Fixed